srcalc/src/main/java/gov/va/med/srcalc/service/DefaultAdminService.java

```diff
@@ -63,11 +64,37 @@ public AbstractVariable getVariable(final String keyName)
         return var;
     }
     
+    /**
+     * Returns true if there is already a different variable with the same key.
+     * @param variable
+     * @return
+     */
+    private boolean keyAlreadyExists(final AbstractVariable variable)
+    {
+        final AbstractVariable existingVar = fVariableDao.getByKey(variable.getKey());
+        fLogger.debug("Existing variable with key {}: {}", variable.getKey(), existingVar);
+        // If there is an existing var with the same key and it is not actually
+        // the same variable, then we have a conflict.
+        return (existingVar != null && variable.getId() != existingVar.getId());
```

Comment:
In the case of saving a variable, wouldn't checking for null be the only thing we have to do?